### ISSUE TITLE
Change examples to presets

### ISSUE THREAD

**Issue by jonniebigodes** [redwoodjs/graphql#288]
I've been tracking the project and so far i'm loving it. I've seen that you have Storybook in mind already and i'm really hoping that it becomes first class and with that i would love to take it out for a spin and expand the tutorial to include this aswell. But i was checking the readme and the tutorial and it would be a nice feature to include in a future release the inclusion of options to the user. And by options i mean the examples you have mentioned. Probably move them into their own repos and allow the initialization process to use them.

Basically if the user issues `yarn create redwood-app ./redwoodblog`  it wil default to the de facto setup. But if the user issues `yarn create redwood-app  todo ./todos` it would initialize a new project  based on that and the same for other existing examples and hopefully giving the option to the community to expand this and allow them to create their own templates.

With that i believe that it could mean more adoption and also allow the user/community to learn the building blocks of the framework.

Feel free to provide feedback

**Comment by thedavidprice** [comment]
Hi @jonniebigodes Interesting idea indeed. And just to make sure you know, the examples currently _are_ in their own public repos and the Example Blog deployed on Netlify (_note: some of these need a little TLC updating to most recent versions_):
### Example Blog
- deployed: https://redwood-example-blog.netlify.com
- repo: https://github.com/redwoodjs/example-blog
### Example Todo
- repo: https://github.com/redwoodjs/example-todo
### Example Invoice
- repo: https://github.com/redwoodjs/example-invoice

Run `git clone …` with any of the above if you want to take ‘em for a spin.

## Discussion: Adding more templates as bootstrap options
It’s definitely possible to add an option/flag to the `yarn create` bootstrap, which most likely would do the same thing as a `git clone`.

The executable behind our `yarn create redwood-app` is here: https://github.com/redwoodjs/redwood/tree/master/packages/create-redwood-app

Very interested in hearing from other about this:
- did you already know examples were available?
- if you did, have you tried the examples locally… why or why not?

Maybe most importantly, if we created a kind of “marketplace” of examples and templates, would people be interested in contributing?

**Comment by jonniebigodes** [comment]
@thedavidprice regarding your comment above. 

I fumbled fingered (pardon the bad pun) and forgot to check the links properly. For that i'm sorry (to much coffee......).

I'm going to clone the examples during this weekend and take a look at them in more detail.

Regarding this:

> Maybe most importantly, if we created a kind of “marketplace” of examples and templates, would people be interested in contributing?

I'm inclined to say yes, but it all depends on spreading the word out and the traction obtained. I'm up for doing my part. So far i'm liking the framework alot. The concepts you put into play are looking really nice. 

Now and if one thing that this ecossystem has taught me is that the possibilities are endless. And with that i can see templates/examples ranging from portfolios, to a C.V (curriculum vitae) and so on.

**Comment by thedavidprice** [comment]
Well, let's do keep this conversation rolling as things progress in your exploration. It's really easy to imagine an MVP marketplace using our existing forum --> we're trying to do this a bit with the "Show and Tell" section (for example). And I know we could help promote. But, again, no need to think about how to jump from A to Z at this time. Let's just noodle on some next steps that validate and build momentum here.

Thanks again!

**Comment by jtoar** [comment]
Linking issue https://github.com/redwoodjs/redwood/issues/2218 and PR https://github.com/redwoodjs/redwood/pull/2324; they're somewhat similar to the conversation here, but have more of a contributing angle.

**Comment by jtoar** [comment]
@KrisCoulson recently mentioned that he'd love to see something like:

```
yarn create redwood-startup my-startup
```

Having a select set of presets is something that we definitely want to explore (after v1).

**Comment by jtoar** [comment]
This came up recently on the Discord. I believe it was @mattmurph9 who asked? Copy pasting here:

> Is there a way to change the template or specify a template to use when creating an app using yarn create redwood-app? I'd like something like yarn create redwood-app --some-template and my app be generated with some different template code, maybe some base data model,  already created frontend pages, something like that

The use case was multiple web apps, so maybe this could be solved via multiple sides instead, and just sharing components:

> My use case is like this: I am creating multiple web applications that have some things in common. Same login page and authentication flow, same topbar, part of the data model is always the same. I'd like to build a template with:

But at least two other users joined the discussion, so it seems like there's still some real interest here.
